contracts/Removal.sol

```diff
@@ -836,20 +836,18 @@ contract Removal is
     for (uint256 i = 0; i < countOfRemovals; ++i) {
       uint256 id = ids[i];
       uint256 amount = amounts[i];
-      if (!_isValidTransfer({amount: amount, to: to})) {
+      if (
+        !_isValidTransferAmount({amount: amount, to: to}) ||
+        (!isValidTransfer &&
+          to != RemovalIdLib.supplierAddress({removalId: id}))
+      ) {
         revert ForbiddenTransfer();
       }
       if (to == market) {
         _currentMarketBalance += amount;
-      }
-      if (from == market) {
+      } else if (from == market) {
```

Comment:
this is the only thing that should be functionally different. It shouldn't ever be possible to have both the to and from address be the market contract. If that's not the case I'll need to change this

Reply:
even if to and from were able to be the same address and a transfer successfully executed in that way, would that ever functionally matter? it would still never be able to create an overly granular balance because no balance would change